src/Orleans.Core/Core/StateStorageBridge.cs

```diff
@@ -27,12 +28,14 @@ public string Etag
             get { return grainState.ETag; }
         }
 
-        public StateStorageBridge(string name, GrainReference grainRef, IStorageProvider store, ILogger logger)
+        public StateStorageBridge(string name, GrainReference grainRef, IGrainStorage store, ILoggerFactory loggerFactory)
```

Comment:
do we need a `LoggerFactory` or just a logger?

Reply:
Oh, we need a logger factory because we want the logging associated with the provider type, which we can't resolve in the constructor.